### PR TITLE
refactor(listitem): create generic listitem component

### LINKED ISSUE #206: Lag felles listItem komponent
Erstatt dagens NewsListItem, EventListItem og JobPostItem. Bruk den Paper-komponenten som skal lages

### LINKED ISSUE #227: Fikse CSS på karriere-siden (Safari)
Deler av komponenten på annonse-siden får en litt merkelig variant av CSS. Dette skjer kun på safari av de jeg har testet. Se vedlagt bilde

Skjermbilde 2020-09-09 kl  15 11 59

### PR DESCRIPTION
**Pull request by olros**
# refactor(listitem): create generic listitem component
Lint, test and build
## Description

Comments/issues/screenshots:
closes #206 
closes #227 

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] PR name is descriptive
- [ ] The PR includes the link to the relevant Trello task
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

- [x] `yarn build` was run locally without errors
- [x] `yarn test` has passed locally and any fixes were made for failures
- [x] `yarn format` was run locally and fixes were made for new errors

## Pull request type

Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: src/components/miscellaneous/ListItem.js [hunk 1]
```diff
@@ -89,51 +82,59 @@ const styles = (theme) => ({
...
-const getDate = (date) => {
-  return date.date() + ' ' + getMonth(date.month()) + ' ' + date.year() + ' - kl. ' + date.format('HH:mm');
-};
-
-const NewsListItem = (props) => {
-  const { classes, data } = props;
-  const src = data.image ? data.image : TIHLDELOGO;
-  const start = moment(data.created_at, ['YYYY-MM-DD HH:mm'], 'nb');
+const ListItem = ({ title, link, expired, img, imgAlt, info }) => {
+  const classes = useStyles();
```
Comment (olros) [inline src/components/miscellaneous/ListItem.js:111]: Alle filer bør endres til å bruke styles på denne måten med hooks

File: src/components/miscellaneous/ListItem.js [hunk 2]
```diff
@@ -1,45 +1,38 @@
...
     },
   },
   src: {
-    objectFit: 'contain',
-    height: '100%',
+    objectFit: 'cover',
+    height: 160,
     width: '40%',
     maxWidth: 250,
     borderRadius: theme.sizes.border.radius,
```
Comment (olros) [inline src/components/miscellaneous/ListItem.js:38]: Kan vurdere å legge inn en default bakgrunns-farge for png'er slik at logoer ikke forsvinner i darkmode slik som på karriere i dag:
image

Må bare finne en god farge

File: src/components/miscellaneous/ListItem.js [hunk 3]
```diff
@@ -89,51 +82,59 @@ const styles = (theme) => ({
...
+  link: PropTypes.string.isRequired,
+  expired: PropTypes.bool,
+  img: PropTypes.string,
+  imgAlt: PropTypes.string,
+  info: PropTypes.arrayOf(
+    PropTypes.shape({
+      icon: PropTypes.object.isRequired,
+      label: PropTypes.string.isRequired,
+    }),
+  ),
```
Comment (olros) [inline src/components/miscellaneous/ListItem.js:137]: Kan vurdere å legge til en `imgContain`-prop som setter `objectFit: contain` i bildet for å sikre at hele bildet vises på for eks karriere siden

File: src/components/miscellaneous/ListItem.js (hunk 1 above)
Comment (Zenjjim) [reply src/components/miscellaneous/ListItem.js:111]: Helt enig, blir mye mer leselig v.h.a hooks

File: src/components/miscellaneous/ListItem.js [hunk 4]
```diff
@@ -89,51 +82,59 @@ const styles = (theme) => ({
...
-  return date.date() + ' ' + getMonth(date.month()) + ' ' + date.year() + ' - kl. ' + date.format('HH:mm');
-};
-
-const NewsListItem = (props) => {
-  const { classes, data } = props;
-  const src = data.image ? data.image : TIHLDELOGO;
-  const start = moment(data.created_at, ['YYYY-MM-DD HH:mm'], 'nb');
+const ListItem = ({ title, link, expired, img, imgAlt, info }) => {
+  const classes = useStyles();
+  const src = img ? img : TIHLDELOGO;
```
Comment (Zenjjim) [inline src/components/miscellaneous/ListItem.js:112]: `const src = img || TIHLDELOGO`

Comment (Zenjjim) [reply src/components/miscellaneous/ListItem.js:112]: eller `??`

File: src/components/miscellaneous/ListItem.js [hunk 5]
```diff
@@ -89,51 +82,59 @@ const styles = (theme) => ({
...
+    <MaterialListItem button className={classes.root} component={Link} to={link}>
+      <img alt={imgAlt || title} className={classNames(classes.src, expired ? classes.filter : '')} src={src} />
       <Grid className={classes.content} container direction='column' wrap='nowrap'>
-        <Typography className={classes.title} variant='h5'>
-          <strong>{data.title}</strong>
+        <Typography className={classNames(classes.title, expired ? classes.expired : '')} variant='h5'>
+          <strong>{title}</strong>
         </Typography>
-        <InfoContent icon={<Date className={classes.icon} />} label={getDate(start)} />
+        {info && info.map((item, index) => <InfoContent icon={item.icon} key={index} label={item.label} />)}
```
Comment (Zenjjim) [inline src/components/miscellaneous/ListItem.js:120]: trenger du `info &&`, er vel bedre å sjekke `info?.map`

File: src/containers/Events/index.js [hunk 6]
```diff
@@ -224,7 +232,21 @@ function Events(props) {
                 <Grow in={!isFetching}>
                   <div className={classes.list}>
                     <Pageination nextPage={getNextPage} page={nextPage}>
-                      {events && events.map((value, index) => <EventListItem data={value} key={value.id} />)}
+                      {events &&
```
Comment (Zenjjim) [inline src/containers/Events/index.js:235]: samme `event?.map`

File: src/containers/NewLanding/components/CalendarListView.js [hunk 7]
```diff
@@ -51,9 +58,23 @@ function CalendarListView({ classes, events, isLoading }) {
     eventList =
       events && events.length > 0 ? (
         <React.Fragment>
-          {events.map((eventData, index) => {
+          {events.map((event, index) => {
             if (index < eventsToDisplay) {
-              return <EventListItem data={eventData} key={index} />;
+              // return <ListItem data={eventData} key={index} />;
```
Comment (Zenjjim) [inline src/containers/NewLanding/components/CalendarListView.js:63]: bruh

File: src/containers/News/index.js [hunk 8]
```diff
@@ -126,7 +132,17 @@ const News = (props) => {
               <Grow in={!isLoading}>
                 <div className={classes.list}>
                   <Pageination nextPage={getNextPage} page={nextPage}>
-                    {displayedNews && displayedNews.map((value) => <NewsListItem data={value} key={value.id} />)}
+                    {displayedNews &&
+                      displayedNews.map((newsItem) => (
```
Comment (Zenjjim) [inline src/containers/News/index.js:136]: same

File: src/containers/Profile/components/ProfileEvents.js [hunk 9]
```diff
@@ -36,9 +42,22 @@ function ProfileEvents(props) {
   return (
     <div className={classes.wrapper}>
       {events &&
-        events.map((eventData, key) => {
-          if (eventData.expired === false) {
-            return <EventListItem data={eventData} key={key} />;
+        events.map((event) => {
+          if (event.expired === false) {
```
Comment (Zenjjim) [inline src/containers/Profile/components/ProfileEvents.js:46]: `!event.expired`

**Review by Zenjjim — CHANGES_REQUESTED** [review]

File: src/containers/NewLanding/components/CalendarListView.js (hunk 7 above)
Comment (olros) [reply src/containers/NewLanding/components/CalendarListView.js:63]: ahaha

**Review by Zenjjim — APPROVED** [review]

**Pull request merged**